app/scripts/services/DataRepository.js

```diff
@@ -179,7 +184,8 @@ var DataRepository = function (recordSets) {
     return null;
   };
 
-  this.insert = function insert(collectionName, record, successMsg, callback) {
+  this.insert = function insert(collectionName, record, successMsg, callback, recordId, currentState) {
```

Comment:
You are completely right! I've checked, and they are always the same. I found that I can just use "record" in place of "currentState" - there was no need to create a copy of it, which is nice because it's simpler.